### DISCUSSION TITLE
How to creat a box with vertex color?

### DISCUSSION THREAD

**Discussion by Bardreamaster** [zauberzeug/nicegui#786] [Q&A] [answered]
### Question

How to creat a box with vertex color?

While I'm creating several boxes in a `ui.scence()` closed to other boxes nearby, these box will be look like a whole "box" instead some separate boxes cause there's no vertex on box to identify different box.

![image](https://user-images.githubusercontent.com/44565280/232675851-58b5aebf-974e-4f79-b1ac-b5a677e92ff8.png)

Is there any way to make vertex on? I've saw there's a [vertexColors](https://threejs.org/docs/index.html#api/en/materials/Material.vertexColors) parameter in three.js.

Thanks so much.

**Comment by falkoschindler** [comment, marked as answer]
Hi @Bardreamaster, that's a great idea!

I just looked into the `vertexColors` parameter and tried to find out, what it exactly does. It looks like it doesn't draw vertices (i.e. corner points) with an individual color, but colors the face with a kind of interpolation of different vertex colors. [Here](http://benchung.com/vertex-colors-three-js/) is an example illustration. This would be quite a change to the current API, since we're currently not even supporting multiple colors per object. And I don't think it is what you're looking for.

But as an alternative, one way to visually separate individual objects could be a combination with wireframes:
```py
with ui.scene() as scene:
    with scene.group().move(x=-0.5, z=0.5):
        scene.box()
        scene.box(wireframe=True).material(color='gray').scale(1.001)
    with scene.group().move(x=0.5, z=0.5):
        scene.box()
        scene.box(wireframe=True).material(color='gray').scale(1.001)
```
![Screenshot 2023-04-20 at 16 55 06](https://user-images.githubusercontent.com/5767091/233405725-2b74a57a-fa58-4edb-8b35-ff008a7f3470.png)

Here I'm grouping two boxes each for easier positioning. And to avoid flickering (due to both boxes having the same camera distance) I'm scaling the wireframes by 0,1% so that the wireframe edges are "outside" of the solid cubes.

What do you think?

**Reply by Bardreamaster** [reply]
@falkoschindler  Thanks a lot. I've also tried to draw a box and a wireframe at the same position with the same size, it looks good !
 
![image](https://user-images.githubusercontent.com/44565280/233528364-c34be77b-8369-4ee7-b6d5-9f5771077827.png)
